src/server/project.ts

```diff
@@ -2046,6 +2046,8 @@ namespace ts.server {
     export class ConfiguredProject extends Project {
         /* @internal */
         configFileWatcher: FileWatcher | undefined;
+        /* @internal */
+        private extendedConfigFileWatchers: ESMap<string, FileWatcher> | undefined;

```

Comment:
Instead of per project we want this to be for all projects so we are watching extended config shared by multiple projects is watched only once.

Reply:
That makes sense, but what about the situation where multiple projects have differing watch options?

Reply:
This is project service wide so this will use `this.hostConfiguration.watchOptions` instead of project specific ones

Reply:
Implemented a shared watcher in (36c29cb) using the host configuration's watch options.